spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/LocalApiKeyManager.java

```diff
@@ -73,16 +72,15 @@ public ApiKey findApiKeyFromLocalApiKeyList(String apiKeyValue) {
         }
         Optional<ApiKey> optionalApiKey = localApiKeyList.stream().filter(
                 apiKeyFromLocalList -> apiKeyFromLocalList.getKeyValue().equals(apiKeyValue)).findFirst();
-        return extractOptionalApiKeyAndLogPossibleErrors(optionalApiKey, apiKeyValue);
+        return extractOptionalApiKey(optionalApiKey);
     }
 
-    private ApiKey extractOptionalApiKeyAndLogPossibleErrors(Optional<ApiKey> optionalApiKey, String apiKeyValue) {
+    private ApiKey extractOptionalApiKey(Optional<ApiKey> optionalApiKey) {
```

Comment:
Why OptionalAPIKey?

Reply:
The findFirst() method of Stream returns an Optional<...>.

Reply:
Ok, and can't you use `optional.get()`, I think that way you would get an APIKey directly

Reply:
The problem is that if get() is used on an empty Optional a NoSuchElementException is thrown.

Reply:
Ok, and can't you use a ifPresent()?
My point here is that on websites I checked, people always comment that Optional should never be used as an argument on a method.
Maybe instead of `.get()` you could try `ifPresent()`?

Reply:
Good point. In the latest version the Optional is not used as an argument anymore.

Reply:
Ok, then I consider this solved